src/SoftRobots/component/constraint/model/CableModel.inl

```diff
@@ -167,7 +167,8 @@ void CableModel<DataTypes>::bwdInit()
 
     Real cableLength = getCableLength(positions.ref());
     Real initialCableLength = getCableLength(restPositions.ref());
-    d_cableInitialLength.setValue(initialCableLength);
+    if (d_cableInitialLength.getValue() == 0)
```

Comment:
Now that I'm looking at it more carefully why this test? A length of 0 could also be wanted by the user...

A clean implementation would be to remove the `d_cableInitialLength.setReadOnly(true);
` from CableModel. 
And in `bwdInit()` put something like: 

```
if (!d_cableInitialLength.isSet()) // not defined by user
{ 
     Real initialCableLength = getCableLength(restPositions.ref());
     d_cableInitialLength.setValue(initialCableLength);
}
```
And don't forget to explain the behavior in the data info:

`d_cableInitialLength(initData(&d_cableInitialLength, Real(0.0), "cableInitialLength","This value can be defined by the user. If not, it will correspond to the length at the beginning of the simulation."))`

However, I'm surprised it works without removing the `readOnly`... It shouldn't... Otherwise, what's the point of using it.

Reply:
I agree!